spec/knapsack_pro/runners/queue/rspec_runner_spec.rb

```diff
@@ -1,536 +1,282 @@
-          expect(rspec_core_runner).to receive(:run).with($stderr, $stdout).and_raise SystemExit
-          expect(KnapsackPro::Hooks::Queue).to receive(:call_before_subset_queue)
-          allow(KnapsackPro::Hooks::Queue).to receive(:call_after_subset_queue)
-          allow(KnapsackPro::Hooks::Queue).to receive(:call_after_queue)
-          allow(KnapsackPro::Formatters::RSpecQueueSummaryFormatter).to receive(:print_exit_summary)
-          expect(Kernel).to receive(:exit).with(1)
-        end
+      context 'when has a format option' do
+        let(:args) { ['--color', '--format', 'd'] }
 
-        it 'does not call #rspec_clear_examples' do
-          expect(described_class).not_to receive(:rspec_clear_examples)
-          expect { subject }.to raise_error SystemExit
+        it 'returns args' do
+          expect(subject).to eq ['--color', '--format', 'd']
         end
+      end
+    end
 
-        it 'logs the exception' do
-          expect(KnapsackPro).to receive(:logger).once.and_return(logger)
-          expect(logger).to receive(:error).with("Having exception when running RSpec: #<SystemExit: SystemExit>")
-          expect { subject }.to raise_error SystemExit
-        end
+    describe '#args_with_default_options' do
+      let(:args) { ['--color', '--format', 'documentation'] }
+      let(:test_dir) { 'spec' }
 
-        it 'calls #print_exit_summary' do
-          expect(KnapsackPro::Formatters::RSpecQueueSummaryFormatter).to receive(:print_exit_summary)
-          expect { subject }.to raise_error SystemExit
-        end
+      subject { function_core.args_with_default_options(args, test_dir) }
 
-        it 'calls #call_after_subset_queue and #call_after_queue' do
-          expect(KnapsackPro::Hooks::Queue).to receive(:call_after_subset_queue)
-          expect(KnapsackPro::Hooks::Queue).to receive(:call_after_queue)
-          expect { subject }.to raise_error SystemExit
-        end
+      it 'adds default options and Knapsack Pro formatters' do
+        expect(subject).to eq [
+          '--color',
+          '--format', 'documentation',
+          '--default-path', 'spec',
+          '--format', 'KnapsackPro::Formatters::TimeTracker',
+        ]
       end
     end
 
-    context "when test files don't exist" do
-      let(:test_file_paths) { [] }
-
-      context 'when all_test_file_paths exist' do
-        let(:all_test_file_paths) { ['a_spec.rb'] }
-        let(:logger) { double }
+    describe '#log_rspec_command' do
+      let(:args) { ['--format', 'documentation'] }
+      let(:test_file_paths) { ['a_spec.rb', 'b_spec.rb'] }
 
-        before do
-          described_class.class_variable_set(:@@used_seed, used_seed)
+      subject { function_core.log_rspec_command(args, test_file_paths, type) }
 
-          expect(KnapsackPro).to receive(:logger).twice.and_return(logger)
+      context 'when there are no test file paths' do
```

Comment:
I'm not sure about that one. Why do we have to couple test description with the variable name?

The similar change I applied for `spec opts => spec_opts` in another test. There, there is no other choice because it's based on `ENV['SPEC_OPTS']` which is a constant name.

Here `test_file_paths` could be renamed to something else. On the higher level we care about test file paths no matter the variable name.

Reply:
It's subjective, feel free to discard.

I use it because it makes easier for me when scanning code that the sentence is related to a specific variable and not an abstraction.